utils/litemap/src/map.rs

```diff
@@ -16,12 +16,13 @@ use core::ops::{Index, IndexMut};
 /// See the [module level documentation][super] for why one should use this.
 ///
 /// The API is roughly similar to that of [`std::collections::BTreeMap`].
+#[allow(clippy::exhaustive_structs)]
 #[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
 #[cfg_attr(feature = "yoke", derive(yoke::Yokeable))]
 pub struct LiteMap<K: ?Sized, V: ?Sized, S = alloc::vec::Vec<(K, V)>> {
-    pub(crate) values: S,
-    pub(crate) _key_type: PhantomData<K>,
-    pub(crate) _value_type: PhantomData<V>,
+    pub values: S,
```

Comment:
Issue: We shouldn't make this `pub`, because it could allow clients to break the sorting invariant. Instead, you could add an associated `const EMPTY` to `Store` or `StoreMut` and then use it in a `pub const LiteMap::new_empty()` function.

Reply:
Another way to do this without requiring that the type is const-constructible would be to expose a function like

```rust
impl<K, V, S> LiteMap<K, V, S> {
    pub const fn new_with_empty_store(store: S) -> Self {
        if !store.lm_is_empty() {
            panic!("Store must be empty");
        }
        Self { store, ... }
    }
}
```

But I think `const EMPTY` is more elegant if you can get it to work. It's fine to put it on another trait.

Reply:
Introduced a new trait `StoreConstEmpty` with an associated const.